front/lib/api/assistant/agent.ts

```diff
@@ -0,0 +1,154 @@
+import { Authenticator } from "@app/lib/auth";
+import { generateModelSId } from "@app/lib/utils";
+import {
+  AgentActionConfigurationType,
+  AgentConfigurationStatus,
+  AgentConfigurationType,
+  AgentMessageConfigurationType,
+} from "@app/types/assistant/agent";
+import {
+  AssistantAgentActionType,
+  AssistantAgentMessageType,
+  AssistantConversationType,
+} from "@app/types/assistant/conversation";
+
+import {
+  RetrievalDocumentsEvent,
+  RetrievalQueryEvent,
+} from "./actions/retrieval";
+
+/**
+ * Agent configuration.
+ */
+
+export async function createAgentConfiguration(
+  auth: Authenticator,
+  {
+    name,
+    pictureUrl,
+    action,
+    message,
+  }: {
+    name: string;
+    pictureUrl?: string;
+    action?: AgentActionConfigurationType;
+    message?: AgentMessageConfigurationType;
+  }
+): Promise<AgentConfigurationType> {
+  return {
+    sId: generateModelSId(),
+    name,
+    pictureUrl: pictureUrl || null,
```

Comment:
nit: not really important here, but may be a good practice to always use `?? null` instead of `||`. Not crucial for strings (we'll just turn `""` into `null`), but for numbers we might change a `0` into `null` (and for booleans we'd overwrite `false`)